src/gar.lua

```diff
@@ -1111,6 +1112,19 @@ function gar.allowDelegates(delegates, gatewayAddress)
 	}
 end
 
+function gar.isEligibleForArNSDiscount(from)
+	local gateway = gar.getGateway(from)
```

Comment:
Could introduce `getGatewayUnsafe` and not suffer the compute and memory costs of cloning all delegate stakes and vaults since this is a read-only operation.

Reply:
8320fc7